### ISSUE TITLE
calling zip.file with a stream, doesn't stream the data, it front-loads it.

### ISSUE THREAD

**Issue by funkjunky** [Stuk/jszip#502]
The whole point of passing a stream as a file is to allow it to be done in chunks, so I can pass it any large file.

I need the stream to only process what data it is currently zipping, so I can throw out the memory when im done.

**Comment by funkjunky** [comment]
This is blocking for me, so i'm going to fork.

**Comment by funkjunky** [comment]
After half a day of searching, I found the accumulator = true and turned it off, and now things pause midfile properly.

However, somewhere in the library all fo the data is accumulating still, because chrome gives a memory out of bounds error at 2GB. The stack trace doesn't reveal anything.

Where are you saving all of this data in your library?

**Comment by funkjunky** [comment]
Nevermind, it was my code that had the memory leak. Apparently I needed to ```delete``` my async iterators next values, because they were accumulating.

Do you really need to have the accumulator when working with streams? Because it defeats the purpose of using streams.

**Comment by funkjunky** [comment]
Ok, the zip was corrupted when i did that... dagnammit... So I'm still trying to figure this out.

**Comment by funkjunky** [comment]
Finally I got everything working correctly. I didn't need the fork. I just needed to set ```{ streamFiles: true }``` for the passed options.

As comments in the code pointed out, the stream doesn't immediately pause, but it does eventually pause. The trick for me, was to not call resume till the stream ACTUALLY paused.

Doing this is tricky and depends on your situation. In my case I queued up promises to handle the data events, then a generator would consume those promises. The generator was ALWAYS calling resume on the stream. Instead I told it to wait till we had consumed all of the promises before calling resume again. With this, the library tends to get ahead by tens of MB, but then truly pauses and my code is then able to catch up.
